### PR TITLE
Graph: Fix when clicking a plot on a touch device we won't display the annotation menu

### LINKED ISSUE #19681: Grafana pop-up annotations instead of values on iPad when I touch a graph
**What happened**:
I'm trying to use Grafana on my iPad however when I touch a graph, I cannot see the values. Values show instantly for a sec then "annotation" bar pops-up. 

**What you expected to happen**:
I expect grafana to show the values when I touch on a graph.

**How to reproduce it (as minimally and precisely as possible)**:
Go to a grafana bar on an iPad and touch a graph.

**Anything else we need to know?**:
Could that be a CSS issue?

**Environment**:
- Grafana version: 6.4.1
- Data source type & version: Influxdb
- OS Grafana is installed on: Debian 9
- User OS & Browser: Safari + iPad iOS13
- Grafana plugins: No additional plugins.
- Others:

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
Hopefully it makes it easier to see different values of a plot on a mobile device.

**Which issue(s) this PR fixes**:
Fixes #19681

**Special notes for your reviewer**:
I know that we shouldn't update a vendor package since it makes it harder to do an upgrade. Had a quick sync with @torkelo about this and in this case it is okay since the jquery.plot library is inactive.

### CODE REVIEW

File: public/app/plugins/panel/graph/graph_tooltip.ts [hunk 1]
```diff
@@ -172,6 +172,16 @@ export default function GraphTooltip(this: any, elem: any, dashboard: any, scope
     appEvents.emit(CoreEvents.graphClicked, { pos: pos, panel: panel, item: item });
   });
 
+  elem.bind('plotleave', () => {
+    if (panel.tooltip.shared) {
```
Comment (hugohaggmark) [inline public/app/plugins/panel/graph/graph_tooltip.ts]: Nit: maybe use quick exit?
```suggestion
    if (!panel.tooltip.shared) { return; }
```

File: public/vendor/flot/jquery.flot.js [hunk 2]
```diff
@@ -3052,6 +3054,42 @@ Licensed under the MIT license.
...
+            if (original.changedTouches.length === 0) {
+                return e;
+            }
+
+            var touch = original.changedTouches[0];
+            e.pageX = touch.pageX;
+            e.pageY = touch.pageY;
+
+            return e;
+        }
```
Comment (hugohaggmark) [inline public/vendor/flot/jquery.flot.js]: Nit: Is this code copied from somewhere? Maybe add source in comment if copied?

**Review by hugohaggmark — COMMENTED** [review]
Looks good, left small nits that you might want to consider. I tried this on a Samsung Tablet and it works, that is the `add annotation` doesn't appear any longer only the graph tooltip. But how to add an annotation now is another question, maybe it's not supported on mobile devices is enough for now?

File: public/vendor/flot/jquery.flot.js [hunk 3]
```diff
@@ -3052,6 +3054,42 @@ Licensed under the MIT license.
           triggerClickHoverEvent("plotclick", e, function (s) { return s["clickable"] != false; });
         }
 
+        function onTouch(e) {
```
Comment (dprokop) [inline public/vendor/flot/jquery.flot.js]: Would be good to annotate this code with `// grafana addition` as we do for vendor code modifications :)

File: public/vendor/flot/jquery.flot.js (hunk 2 above)
Comment (dprokop) [reply public/vendor/flot/jquery.flot.js]: Agree with @hugohaggmark. Either add link to source or leave some comment with reasoning

Comment (mckn) [reply public/vendor/flot/jquery.flot.js]: Not copied from somewhere :)

**Comment by mckn** [comment]
> Looks good, left small nits that you might want to consider. I tried this on a Samsung Tablet and it works, that is the `add annotation` doesn't appear any longer only the graph tooltip. But how to add an annotation now is another question, maybe it's not supported on mobile devices is enough for now?

Me and @torkelo talked about this and if I understood everything correctly we can skip it for now. But maybe we should create a issue on adding a solution for this so we don't forget about it. What do you think @torkelo ?

File: public/vendor/flot/jquery.flot.js (hunk 2 above)
Comment (mckn) [reply public/vendor/flot/jquery.flot.js]: But maybe I should rename the function a bit to make it more clear?

**Comment by mckn** [comment]
> > Looks good, left small nits that you might want to consider. I tried this on a Samsung Tablet and it works, that is the `add annotation` doesn't appear any longer only the graph tooltip. But how to add an annotation now is another question, maybe it's not supported on mobile devices is enough for now?
> 
> Me and @torkelo talked about this and if I understood everything correctly we can skip it for now. But maybe we should create a issue on adding a solution for this so we don't forget about it. What do you think @torkelo ?

After talking to @hugohaggmark I realized that if you press the tooltip the `add annotation` will appear. So it is possible and feels pretty okay.

File: public/vendor/flot/jquery.flot.js (hunk 2 above)
Comment (hugohaggmark) [reply public/vendor/flot/jquery.flot.js]: Wow great job on that!

File: public/vendor/flot/jquery.flot.js [hunk 4]
```diff
@@ -3052,6 +3054,44 @@ Licensed under the MIT license.
           triggerClickHoverEvent("plotclick", e, function (s) { return s["clickable"] != false; });
         }
 
+        /* Grafana Addon - added to support mobile devices click in plot */
```
Comment (dprokop) [inline public/vendor/flot/jquery.flot.js]:
```suggestion
        // grafana addition - added to support mobile devices click in plot 
```

File: public/vendor/flot/jquery.flot.js [hunk 5]
```diff
@@ -3052,6 +3054,44 @@ Licensed under the MIT license.
...
+            if (!eventHolder.is(e.target) && eventHolder.has(e.target).length === 0) {
+                triggerClickHoverEvent("plotleave", e, function (s) { false; });
+                return;
+            }
+
+            onMouseMove(mapFromTouchEvent(e));
+            e.preventDefault();
+        }
+
+        /* Grafana Addon - added to support mobile devices and mapping touch event to click event structure */
```
Comment (dprokop) [inline public/vendor/flot/jquery.flot.js]:
```suggestion
        // grafana addition - added to support mobile devices and mapping touch event to click event structure 
```

**Review by dprokop — APPROVED** [review]

File: public/vendor/flot/jquery.flot.js (hunk 4 above)
Comment (mckn) [reply public/vendor/flot/jquery.flot.js]: yupp, will update :)

**Pull request merged**